site/content/docs/developer/building-zap-with-eclipse.md

```diff
@@ -0,0 +1,65 @@
+---
+title: "Building ZAP with Eclipse"
+description: "This guide will get you started with building and running ZAP in the Eclipse IDE."
+tags: 
+- guide
+- tutorial
+type: page
+date: "2021-03-18"
+---
+
+If you've want to make changes to ZAP using the Eclipse IDE then you are in the right place.
+
+## Preparation
+You will need to have followed the [Quick Start Guide to Building ZAP](../quick-start-build/) and installed [Eclipse](https://www.eclipse.org/).
+
+## Import the ZAP Repos
+In Eclipse select the 'File' menu, then 'Import...', and then 'Gradle' &#8594; 'Existing Gradle Project':
+
+![Eclipse Gradle Import](/img/docs/developer/eclipse-import.png)
+
+In the following dialog enter the full directory name for one of the ZAP repos you cloned before and click on 'Finish'.
+
+Repeat this process until you have imported all of the ZAP repos.
+
+## Running ZAP
+In the Eclipse Project Explorer navigate to:
+* zaproxy / zap / src / main / java / org / zaproxy / zap
+
+Right click `ZAP.java` and select 'Run As' &#8594; 'Java Application'.
+
+ZAP will now start but it will not be set up quite right and may show warning dialogs, so close it and go back to Eclipse.
+
+Select the 'Run' menu, then 'Run Configurations...' and then the `ZAP` configuration.
+
+Select the 'Arguments' tab, change the 'Working directory' from 'Default' to 'Other' and enter: `${workspace_loc:zaproxy/zap/src/main/dist/}`
+
+![Eclipse Gradle Import](/img/docs/developer/eclipse-configs.png)
+
+Run this configuration and ZAP should now start with all of the add-ons correctly configured.
+
+## Running Gradle Tasks
+You can run the ZAP Gradle tasks from the command line or from within Eclipse, the end result will be the same.
+
+Eclipse has a 'Gradle Tasks' tab that lists all of the tasks available to you ordered by project. 
+Note that the add-ons in zap-extensions all have their own set of tasks in addition to the top level ones.
+
+## Tips and Tricks
+
+### Refresh Gradle Project
+
+Sometimes Eclipse will unexpected fail to compile ZAP code. This can occur if you switch between git branches which have different dependencies.
```

Comment:
```suggestion
Sometimes Eclipse will unexpectedly fail to compile ZAP code. This can occur if you switch between git branches which have different dependencies.
```